19575 Further styling of classic Work Package views

myabc opened the pull request:
:warning: ~~**Includes commits from #2815, Please review, merge first.**~~
## What this PR/WP does:
- applies various styling from the details pane to the classic Work Package views, namely:
  - _attributes key values_
  - _attributes groups_
  - and introduces _attributes tables_
- changes styling for collapsible fieldsets
## What this PR/WP does not do:
- ~~adjust padding around the related work packages/watchers select~~
- adjust paddings
- some requested changes (different arrows, formatting, etc.) in the work package hierarchy table.

https://community.openproject.org/work_packages/19575

![seeded_project_-__817__et_placeat_optio_corrupti_voluptatem_voluptas_consequatur_provident_-_openproject](https://cloud.githubusercontent.com/assets/755/7140845/051e6670-e2ce-11e4-923e-8221353ce148.png)

myabc committed:
Rework classic wp page using attributes groups

myabc committed:
Remove unnecessary spacing in work packages forms

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Rework classic wp page using new attributes styles

Signed-off-by: Alex Coles <alex@alexbcoles.com>

TeatroIO commented:
I've prepared a stage to preview changes. <a href='http://myabc-feature-19575-quick-win-work-packages-page-styling.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com' target='_blank'>Open stage</a> or <a href='http://myabc-feature-19575-quick-win-work-packages-page-styling.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com/teatro.log' target='_blank'>view logs</a>.

myabc committed:
Add separate classes for attributes group controls

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Unify details pane and classic wp page tables

Rework, document as attributes table.

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Specify column widths on classic wp page tables

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
In-line details pane column widths in template

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc commented:
@NobodysNightmare @floriank ready for review

myabc committed:
Living Style Guide: Remove stray .md file

[ci skip]

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Living Style Guide: Fix whitespace (tabs->spaces)

[ci skip]

Signed-off-by: Alex Coles <alex@alexbcoles.com>

NobodysNightmare commented on app/helpers/work_packages_helper.rb:
```diff
@@ -327,22 +327,33 @@ def work_package_show_attributes(work_package)
 
-  def work_package_show_table_row(attribute, klass = nil, attribute_lang = nil, value_lang = nil, &block)
-    klass = attribute.to_s.dasherize if klass.nil?
+  def work_package_show_dd_dt(attribute, css_class = nil, attribute_lang = nil, value_lang = nil, &block)
```
Sorry @myabc, but you are breaking plugins here (at least costs, probably more).

If I see this correctly the method-signature stays compatible, in that case we could use... how is it called, `deprecate_alias` or something?
Just make the old method available and call it deprecated ^^

NobodysNightmare commented on frontend/app/templates/work_packages/tabs/_attachments_table.html:
```diff
@@ -1,4 +1,11 @@
+    <col style="width: 10%">
+    <col style="width: 20%">
+    <col style="width: 15%">
+    <col style="width: 25%">
```
Just asking: Inline style elements don't violate any rules... Like the "never use style tags"-rule or something?

Don't know if that is a thing...

myabc replied:
@NobodysNightmare When there is only one usage, I think inlining styles makes more sense.

NobodysNightmare commented on app/helpers/work_packages_helper.rb:
```diff
@@ -327,22 +327,33 @@ def work_package_show_attributes(work_package)
     group_work_package_attributes work_package_show_attribute_list(work_package)
   end
 
-  def work_package_show_table_row(attribute, klass = nil, attribute_lang = nil, value_lang = nil, &block)
```
Sorry @myabc, but you are breaking plugins here by renaming the method (at least costs, probably more).

If I see this correctly the method-signature stays compatible, in that case we could use... how is it called, `deprecate_alias` or something?
Just make the old method available and call it deprecated ^^

edit: Moved the comment so that it is rendered at the bottom on GitHub... @houndci's comment on the other line made it appear above the "ready for review" line, that might have been confusing ^^

myabc replied:
@NobodysNightmare good catch. It honestly didn't occur to me that this is _part of our (semi-)public plugin API)_, so I didn't grep for usages. In the future, we really need to document the Ruby plugin API.

NobodysNightmare replied:
I initially also thought that adding attributes was more generic. I was just looking for layout issues, but got a 500 instead ^^

myabc replied:
@NobodysNightmare should be fixed in ~~3ff4a1a~~ e625539. I guess I ought to go through all the plugins though :frowning:

Commit:
fixes the add watchers mini-form

The component was not updated to the new form builder and used grid-block
styles which broke with the upgrade to foundation 1.1

The form now uses a choice component (see living style guide)

Also, the cancel button has been removed there is no good way to leave it in
(without extending the choice component) and it is probably not used anyway.

myabc committed:
Correct styles used by #send_notification_option

Use check-box with label styling for a left-aligned check-boxes.

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Adjust classic wp page table widths

* Hide unnecessary Type column in relations table.

myabc committed:
Fix check-box with label padding

Commit:
correct a styling issue in the styleguide

so meta.

[ci skip]

myabc committed:
Living Style Guide: remove unstyled example

[ci skip]

myabc committed:
Living Style Guide: better label+check-box example

[ci skip]

myabc committed:
Move collapsible fieldset toggle to left

* Match toggle icon with menu sidebar.

Signed-off-by: Alex Coles <alex@alexbcoles.com>

floriank commented:
@myabc will merge #2828 into this branch.

myabc committed:
Add deprecated alias for renamed wp helper method

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Merge branch 'fix/19610-fix-watchers' of https://github.com/floriank/openproject into feature/19575-QUICK-WIN-work-packages-page-styling

Signed-off-by: Alex Coles <alex@alexbcoles.com>

Conflicts:
	app/views/watchers/_watchers.html.erb

myabc committed:
Remove border from attributes table header rows

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Reduce attributes group header border

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Fix indent in WorkPackagesHelper

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Move watchers select form below list on wp page

This makes the behaviour consist with relations form.

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Rework wp page relations form to use choice styles

Brings parity with watchers. See fa15c21a.

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Add cancel button back to wp page watchers form

Removed in fa15c21a.

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc commented:
This is _good enough_<sup>TM</sup>

floriank commented:
works for me, we should probably add the clear to the component _at some point_ - i think it's not very nice with the added `div`.

myabc commented:
@floriank yep, definitely .. I also think _selection form_ or _inline form_ or such like might be a better name.

floriank commented:
I'll keep it in mind - I am horrendous when it comes to naming stuff. my first born will probably get a bar code or something.

NobodysNightmare commented:
MÖRTSCHT!

floriank commented:
@NobodysNightmare - it's a soft "T", called a "D" I believe
